File: problems/0739/solution.cpp
#include <vector>
#include <iostream>
using namespace std;

class Solution
{
public:
    vector<int> dailyTemperatures(vector<int> &temperatures)
    {
        int n = temperatures.size();
        vector<int> result(n, 0);
        stack<pair<int, int>> myStack;
        for (int i = 0; i < n; i++)
        {
            while (!myStack.empty() && (temperatures[i] > myStack.top().second))
            {
                result[myStack.top().first] = i - myStack.top().first;
                myStack.pop();
            }
            myStack.push({i, temperatures[i]});
        }
        return result;
    }
};

File: problems/0739/solution.py
from typing import List

class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        myStack = []
        result = [0 for _ in range(n)]
        for idx, t in enumerate(temperatures):
            while len(myStack) != 0 and t > myStack[-1][0]:
                result[myStack[-1][1]] = idx - myStack[-1][1]
                myStack.pop()
            myStack.append((t,idx))
        return result
    
if __name__ == "__main__":
    temperatures = [30,60,90]
    print(Solution().dailyTemperatures(temperatures=temperatures))